scripts/post_install.sh

```diff
@@ -16,24 +16,25 @@ UPDATE_INSTRUCTIONS="yml/update_instructions.yaml"
 AEROGEARCATALOG_REGISTRY="yml/aerogearcatalog_registry.yaml"
 ASB_TEMPLATE_FILENAME="_tmp-asb-template.yaml"
 ASB_DC_NAME=$(oc get dc -o jsonpath='{.items[*].metadata.name}' -n "$ASB_PROJECT_NAME")
-ASB_CONTAINER_COUNT=$(oc get pods -l app="$ASB_PROJECT_NAME" -o jsonpath='{.items[*].spec.containers[*].name}' -n "$ASB_PROJECT_NAME" | wc -w | awk '{$1=$1};1')
 
 # Mobile CRD (mobileclient)
 oc create -f ../deploy/crd.yaml
 oc create clusterrole mobileclient-admin --verb=create,delete,get,list,patch,update,watch --resource=mobileclients
 oc adm policy add-cluster-role-to-group mobileclient-admin system:authenticated
 
 # ASB configuration
-until oc get pods -l app="$ASB_PROJECT_NAME" -n "$ASB_PROJECT_NAME" | grep "$ASB_CONTAINER_COUNT/$ASB_CONTAINER_COUNT"
-do
-    echo "Waiting for ASB to spin up" && sleep 5
-done
 oc get configmap broker-config -n "$ASB_PROJECT_NAME" -o jsonpath='{.data.broker-config}' > $ASB_TEMPLATE_FILENAME
-# Append dockerhub registry target - aerogearcatalog
-$yq m -a -i $ASB_TEMPLATE_FILENAME $AEROGEARCATALOG_REGISTRY
-# Additional changes to broker configuration file
-$yq w -i -s $UPDATE_INSTRUCTIONS $ASB_TEMPLATE_FILENAME
-# Replace double quotes with single quotes
-sed -i.bak -e "s/\"/'/g" $ASB_TEMPLATE_FILENAME
-oc patch configmap broker-config -p "{\"data\":{\"broker-config\": \"$(awk '{printf "%s\\n", $0}' ${ASB_TEMPLATE_FILENAME})\"}}" -n "$ASB_PROJECT_NAME"
-oc rollout latest "$ASB_DC_NAME" -n "$ASB_PROJECT_NAME"
\ No newline at end of file
+if ! $yq r $ASB_TEMPLATE_FILENAME registry.*.org | grep aerogearcatalog &>/dev/null ; then
+    while oc rollout history dc/"$ASB_DC_NAME" -n "$ASB_PROJECT_NAME" | grep Running &>/dev/null
+    do
+        echo "Waiting for ASB to spin up" && sleep 5
+    done
```

Comment:
why do we need to wait for the ASB to be up & running, before we can make the config change?

Reply:
after the config map is changed we need to redeploy the ASB, but this can not be done if deployment is running